lib/ramble/ramble/application.py

```diff
@@ -1426,9 +1449,170 @@ def format_context(context_match, context_format):
+        each experiment's repeats and calculates statistics for each numeric FOM.
+
+        If a FOM is non-numeric, no calculations are performed.
+
+        Statistics are injected into the results file under the base experiment
+        namespace.
+        """
+
+        def is_numeric(value):
+            """Returns true if a fom value is numeric"""
+
+            try:
+                float(value)
+                return True
+            except ValueError:
+                return False
+
+        repeat_experiments = {}
+        repeat_foms = {}
+        first_repeat_exp = ''
+
+        # repeat_experiments dict {repeat experiment namespace: {dict}}
+        # repeat_foms dict {context: {(fom name, units, origin, origin_type): [list of values]}}
+        # origin_type is generated as 'summary::stat_name'
+
+        # Check if this is the base experiment of a set of repeats
+        if not self.repeats[0]:
+            return
+        n_repeats = self.repeats[1]
+
+        base_exp_name = self.expander.experiment_name
+        base_exp_namespace = self.expander.experiment_namespace
+
+        # Create a list of all repeats by inserting repeat index
+        for n in range(1, n_repeats + 1):
+            if '.chain' in base_exp_name:
+                insert_idx = base_exp_name.find('.chain')
+                repeat_exp_namespace = base_exp_name[:insert_idx] + f'.{n}' \
+                    + base_exp_name[insert_idx:]
+            else:
+                base_exp_namespace = self.expander.experiment_namespace
+                repeat_exp_namespace = f'{base_exp_namespace}.{n}'
+            repeat_experiments[repeat_exp_namespace] = {}
+            repeat_experiments[repeat_exp_namespace]['base_exp'] = base_exp_namespace
+            if n == 1:
+                first_repeat_exp = repeat_exp_namespace
+
+        # Create initial results dict since analysis pipeline was skipped for base exp
+        self.results = {'name': base_exp_namespace}
+        self.results['N_REPEATS'] = n_repeats
+        self.results['EXPERIMENT_CHAIN'] = self.chain_order.copy()
+
+        #
+        # If repeat_success_strict is true, one failed experiment will fail the whole set
+        # and statistics will not be calculated
+        # If repeat_success_strict is false, statistics will be calculated for all successful
+        # experiments
+        #
+        repeat_success = False
+        exp_success = []
+        for exp in repeat_experiments.keys():
+            if exp in self.experiment_set.experiments.keys():
+                exp_inst = self.experiment_set.experiments[exp]
+            elif exp in self.experiment_set.chained_experiments.keys():
+                exp_inst = self.experiment_set.chained_experiments[exp]
+            else:
+                continue
+
+            exp_success.append(exp_inst.get_status())
+
+        if workspace.repeat_success_strict:
+            if 'FAILED' in exp_success:
```

Comment:
Should this use `experiment_status` ?

Reply:
It does. ``exp_success`` is a list of the ``experiment_status`` for every repeat in the set. It then applies the strict/loose criteria to determine the status of the repeat set. If strict, any failure causes the set to fail. If not strict, any success causes the set to succeed.

Reply:
Sorry I mean more "it looks like you are using a magic string that I think we have in an enum. Are they the same and if so can we use the enum?"

Reply:
Got it. Since the use of get_status() above creates a list of enum names as strings, I've updated it to check for the enum name. Does this look right?
```
if experiment_status.FAILED.name in exp_success:
```

Reply:
Yea, I think so